src/npctalk.cpp

```diff
@@ -251,6 +251,7 @@ void npc::handle_sound( int priority, const std::string &description, int heard_
             // TODO: Brave NPCs should be less jumpy
             if( ai_cache.total_danger < 1.0f ) {
                 warn_about( "combat_noise", rng( 1, 10 ) * 1_minutes );
+                ai_cache.spos = spos;
```

Comment:
NPCs can process multiple sound events in a single turn.  I suggest making this a vector that gets sorted by position and they investigate the closest noise until all noises are investigated.  If you do that, they should probably ignore noises that are more than 50 tiles away from their guard position.

This might result in NPCs on guard moving around quite a bit, but I don't think that's a bad thing.